packages/database/test/query.test.ts

```diff
@@ -2191,6 +2195,111 @@ describe('Query Tests', () => {
+    const node = getRandomNode() as Reference;
+    await node.set({ foo: 'a', bar: 'b' });
+    const snapshot = await node.get();
+    const val = snapshot.val();
+    expect(val['foo']).to.equal('a');
+    expect(val['bar']).to.equal('b');
+  });
+
+  it('get for removed node returns correct value', async () => {
+    const node = getRandomNode() as Reference;
+    await node.set({ foo: 'a', bar: 'b' });
+    let snapshot = await node.get();
+    let val = snapshot.val();
+    expect(val['foo']).to.equal('a');
+    expect(val['bar']).to.equal('b');
+    await node.remove();
+    snapshot = await node.get();
+    val = snapshot.val();
+    expect(val).to.be.null;
+  });
+
+  it('get while offline is null', async () => {
+    const node = getRandomNode() as Reference;
+    node.database.goOffline();
+    const snapshot = await node.get();
+    expect(snapshot.val()).to.be.null;
+    // Required for test isolation.
+    node.database.goOnline();
+  });
+
+  it('get caches results at path', async () => {
+    const reader = getFreshRepo('db');
+    const writer = getFreshRepo('db');
+
+    await writer.set({ foo: 'bar' });
+    const snapshot = await reader.get();
+    reader.database.goOffline();
+    reader.once('value', snap => {
+      expect(snap.val()).to.equal(snapshot.val());
+    });
+  });
+
+  it('get reads from cache if database is not connected', async () => {
+    const node = getFreshRepo('db');
+    const node2 = getFreshRepo('db');
+    await node2.set({ foo: 'bar' });
+    const prom = new Promise((resolve, _) => {
+      node.on('value', snap => {
+        resolve(snap);
+      });
+    });
+    const onSnapshot = await prom;
+    node.database.goOffline();
+    const getSnapshot = await node.get();
+    node.off();
+    expect(getSnapshot.val()).to.deep.equal((onSnapshot as DataSnapshot).val());
+  });
+
+  it('get above path caches data', async () => {
+    const reader = getFreshRepo('db');
+    const writer = getFreshRepo('db');
+
+    await writer.set({ foo: { bar: 'baz' } });
+    const snapshot = await reader.get();
+    reader.database.goOffline();
+    reader.child('foo/bar').once('value', snap => {
+      expect(snap.val()).to.equal(snapshot.val());
+    });
+  });
+
+  it('get does not cache sibling data', async () => {
+    const reader = getFreshRepo('db');
+    const writer = getFreshRepo('db');
+    await writer.set({ foo: { bar: { data: '1' }, baz: { data: '2' } } });
+    const snapshot = await reader.child('foo/bar').get();
+    expect(snapshot.val().data).to.equal('1');
+    reader.database.goOffline();
+    const prom = new Promise((resolve, reject) => {
+      let done = false;
+      setTimeout(() => {
+        if (done) {
+          return;
+        }
+        done = true;
+        reject();
+      }, 3000);
+      reader.child('foo/baz').once('value', snapshot => {
+        if (done) {
+          return;
+        }
+        done = true;
+        resolve(snapshot);
+      });
```

Comment:
Couldn't you just use `get()` here again since we are offline and it falls back to `once`?